### DISCUSSION TITLE
"Unerreichbar" Geräte reaktivieren

### DISCUSSION THREAD

**Discussion by emufan** [SukramJ/aiohomematic#566] [General]
Derzeit ist es ja so, dass, wenn ich z.B. einen Rollladen herunterlassen will (hier nur Beispiel, gleiches bei Stromaktoren), es in dem Moment sein kann, dass die CCU diesen aus welchen Gründen auch immer nicht erreicht. Die CCU schalte dann eine Servicemeldung und die Integration graut das ganze Gerät aus.

1. Wenn man nichts macht, probiert die CCU periodisch wieder, ob das Gerät doch da ist oder wartet bis es wieder einen periodischen Status an die CCU sendet. Richtig?

2. Dann kommt es auch wieder in die Events an HA und die Integration aktiviert das Gerät wieder. Richtig?

Im konkreten Fall macht der Rollladen so lange nichts und lässt sich nicht bedienen, bis das obige geschieht und das Gerät wieder aktiv und bedienbar wird.

**Aber**: Gehe ich trotz Servicemeldung in der CCU auf das Gerät, kann ich den Bedienknopf **dennoch** drücken und die CCU probiert es anscheinend einfach, den Befehl trotz Servicemeldung zu senden, was bei mir dann auch zu 100% funktioniert und sofort die Servicemeldung löscht, den Rollladen fahren lässt und das Gerät in HA auch wieder aktiviert. 

Das geht in HA aktuell nicht, oder? Und ich muss auf 1 und 2 warten, ohne den Reconnect provozieren/produzieren zu können, oder?

Oder gibt es da doch eine Möglichkeit, die ich in HA nutzen kann, um wie in der CCU das Senden nochmal probieren zu können? Muss es denn sein, dass das Gerät in HA sofort auf unavailable geht, sodass man es nicht mehr bedienen kann, wodurch die Möglichkeit des "nochmaligen Drückens" schon deshalb wegfällt?

**Comment by SukramJ** [comment]
Hast Du gut beschrieben.

> Oder gibt es da doch eine Möglichkeit, die ich in HA nutzen kann, um wie in der CCU das Senden nochmal probieren zu können? 

Service calls gehen auch bei `unavailable`.

> Muss es denn sein, dass das Gerät in HA sofort auf unavailable geht, sodass man es nicht mehr bedienen kann, wodurch die Möglichkeit des "nochmaligen Drückens" schon deshalb wegfällt?

Ich sag erst mal ja, oder hast Du einen besseren Vorschlag.

Um was für ein Gerät handelt es sich konkret?

**Reply by emufan** [reply]
Hier ein HmIP-BROLL. Da ist HmIP bei mir recht instabil. Vielleicht weil ich so viel habe, aber warum die CCU sofort auf Servicemitteilung geht, anstatt es ein paar Millisekunden später nochmal zu probieren, weiß ich aber auch nicht. Das ist aber ja nicht Problem der Integration, was ich verstehe.

> Ich sag erst mal ja, oder hast Du einen besseren Vorschlag.

Also am besten fände ich, wenn das Gerät nicht direkt auf `unavailable` geht, sondern aktiv bleibt. Sodass ich den Knopf nochmal drücken kann, wie ich es wie beschrieben im WebUI ja auch machen kann und womit das Gerät dann direkt reaktiviert wird, denn es ist ja da, nur die CCU agiert halt so doof (s.o.). Dafür kenne ich Dein Framework aber nicht und weiß nicht, ob das nicht möglich wäre?

> Service calls gehen auch bei `unavailable`.

Dann müsste man sich aber die komplette HA Bedienung nachbauen und die Standard cover-entitiy-rows ersetzen. Nebst Nachbau der Iconsteuerung mit card_mod, Verlust des Sliders (bzw. auch dortigem Nachbau per browser_mod more_info, usw. usw. Quasi um das Gleiche zu haben, was "man hätte", wenn das Gerät nicht `unavailable` gehen würde.

Aber ich verstehe schon, dass man dann ein anderes Kriterium für das "wirkliche" `unavailable` haben müsste, hab' dazu aber auch noch keine gute Idee. Oder gibt es einen Status, der quasi `unavailable` ausdrückt, aber das HA-UI weiterhin aktiv lässt?

**Reply by SukramJ** [reply]
Da es sich um ein einzelnes Gerät handelt würde ich nichts ändern.

Vielleicht funktioniert das:
Du könntest auf das Ereignis `homematic.device_availability` reagieren und eine _Dummy_ Aktion auf dem Gerät ausführen.
Das Event wird auch im [blueprint](https://github.com/danielperna84/custom_homematic/blob/devel/blueprints/automation/homematicip_local_persistent_notification.yaml) verwendet.

**Reply by emufan** [reply]
Einzelnes Gerät glaube ich tatsächlich nicht, sondern nach meinem Gefühl viele/alle (?), wenn die CCU gleichzeitig viele Geräte schalten muss und sich dabei verhaspelt oder überfordert ist. Und ja, habe auch schon Pausen eingebaut, usw., was aber wiederum die Gruppen in HA nicht nutzbar/unbrauchbar werden lässt. Zumindest hatte ich es auch schon beim HmIP-BSM.

> Du könntest auf das Ereignis homematic.device_availability reagieren und eine Dummy Aktion auf dem Gerät ausführen.
Das Event wird auch im [blueprint](https://github.com/danielperna84/custom_homematic/blob/devel/blueprints/automation/homematicip_local_persistent_notification.yaml) verwendet.

Werde ich mal probieren. Vielleicht kann die Dummy-Aktion ja auch die Nutzung des neuen Edgecase-Services sein, damit ich hier gar nichts dummymäßig steuen muss. 😉

Oder ich mache mir eine double_tap-Aktion ins UI, die das auf die aktuelle Entitiät ausführt. Mal gucken. Werde berichten, was klappt und was nicht so gut.

**Reply by kwithus** [reply]
Moin, dies Problem habe ich auch öfters. Der HMIP-Broll ist auch bei mir ein häufiger Kandidat, aber ich vermute auch, weil eben im Zweifel oft mehrere Rollläden gleichzeitig angesprochen werden. Ansonsten sind auch meine verbliebenen ‚Nicht-IP‘ HM devices oft betroffen. Bei denen ist es noch ärgerlicher, weil sie selten von sich aus wieder aktiv gehen. Da hilft dann nur ein kurzes bedienen in der WebUI. 
es wäre wirklich toll, wenn die Integration dies abfangen könnte, in dem z.B. der letzte bekannte Zustand oder der letzte Befehl bei ‚unavailable
 Noch ein paar Sekunden nochmals gesendet würde.

**Comment by SukramJ** [comment]
Versucht mal Version 1.16.0.
Hier gibts es einen neuen [Service](https://github.com/danielperna84/custom_homematic#homematicip_localforce_device_availability) der das `available` Attribut überschreibt, wodurch das Gerät inklusive Entitäten wieder als `available` im System erscheint, und somit wieder bedienbar wird. Das dann nicht unbedingt dem realen Zustand, aber das sollte ja jedem klar sein.
Zur einfacheren Verwendung gibt es zusätzlich drei Blueprints.
Der Rest ist den Release Notes und der Readme zu entnehmen.
Ich bitte um Rückmeldung.

**Reply by emufan** [reply]
Das klingt fantastisch. "Leider" hatte ich in den letzten Tagen gar keine Ausfälle. Aber wenn einer vorbeikommt, werde ich das direkt testen.

**Reply by SukramJ** [reply]
Vielleicht einen der Blueprints einrichten, dann sieht man das bei der Auslösung.

**Reply by emufan** [reply]
Klar. Hatte eh schon vorher eine allgemeine Automatisierung per `homematic.device_availability` trigger und nun dort bereits den Devicetyp-`homematicip_local.force_device_availability` Call auch schon eingebaut.

**Comment by kwithus** [comment]
Der neue Service funktioniert hervorragend, Danke!
Habe das blueprint full installiert und hatte heute mehrmals neu gestartet (aus anderen Gründen). Typischerweise Situationen, in denen es zu unavailable kommt. Wurde mir auch in den Benachrichtigungen angezeigt, tatsächlich waren die devices aber sofort wieder erreichbar, die blueprint Automation hatte sich drum gekümmert. 
jetzt fehlt nur noch, dass die Benachrichtigung automatisch gelöscht wird ;-)

**Reply by SukramJ** [reply]
> tatsächlich waren die devices aber sofort wieder erreichbar

Nur zur Klarstellung:
Der Service/Blueprint setzt nur das `available` flag in HA auf `true`. Von Seiten der CCU ist ggf. weiter das `UNREACH` Event gesetzt, d.h. aus CCU Sicht ist das Gerät eigentlich nicht erreichbar. Mag nicht richtig sein, ist aber so.

Ich würde die Bluprints mit device_type bzw. single device empfehlen, um das `available` flag nur bei problematischen Geräten zurückzusetzten. Ansonsten verliert man den Überblick welche Geräte wirklich nicht verfügbar sind.

> jetzt fehlt nur noch, dass die Benachrichtigung automatisch gelöscht wird

Der persistent notification blueprint löscht das Flag, sobald die CCU nicht mehr `UNREACH` sagt, und ist somit eine gute Erganzung.

**Reply by eikowagenknecht** [reply]
Wollte mich nur fix hier bedanken, der Service ist für meine 5 sporadisch angeblich nicht erreichbaren Rolladenaktoren gold wert, um sie dennoch bedienen zu können 😊

**Reply by SukramJ** [reply]
Danke, das hört man gerne.

**Comment by kwithus** [comment]
Leider hat es heute bei einem actor mit der re-Aktivierung überhaupt nicht geklappt.
Der HM-LC-Bl1-FM wurde in HA als nicht verfügbar angezeigt, blueprint für Full und für Single device haben beide nichts geändert und das, obwohl der Autor in der CCU WebUI steuerbar war.
Nur ein Neustart von HA konnte ihn zum Leben erwecken ;-)

**Reply by SukramJ** [reply]
> Nur ein Neustart von HA konnte ihn zum Leben erwecken ;-)

Wie gesagt: Das ist nicht die Aufgabe des Service. Wir ändern nur den Status im HA.

**Comment by kwithus** [comment]
Ok, habe mich falsch ausgedrückt. ‚Am Leben‘ war er eben schon lange (in der CCU) aber in HA wurde er (trotz Auslösen der Blueprint Automation) als nicht verfügbar gezeigt. Erst nach dem Neustart von HA war er wieder verfügbar, davor 5 STD. Lang nicht

Von meinem iPad gesendet

**Reply by SukramJ** [reply]
OK, hört sich jetzt aber anders an.
Beim nächstem mal bitte den Service `homematicip_local.force_device_availability` manuell auslösen.

**Comment by kwithus** [comment]
Ja, sorry, war blöd formuliert. Versuche ich das nächste mal.
Hatte gerade HA nochmal neu gestartet , da das der häufigste Fall ist, dass HM devices nicht erreichbar sind (angeblich), in der Tat kam auch für diesen Aktor die Meldung, aber diesmal klappte es sofort mit den Neusetzen des Status.

Von meinem iPad gesendet

**Comment by Aggro600** [comment]
Ich habe genau das selbe Problem. Gibt es schon eine Lösung?

**Reply by SukramJ** [reply]
In den letzten drei Monaten hat sich dazu nichts mehr getan, und ich denke, dass der Status Quo so in Ordnung ist.

**Reply by Aggro600** [reply]
schade mit openHab und debmatic kannte ich diese Probleme nicht einmal.

**Reply by SukramJ** [reply]
Hast Du eigentlich in dieser Diskussion die anderen Beiträge gelesen? 
Hast Du dir mal die Dokumentation durchgelesen?

**Reply by Aggro600** [reply]
Mit der blueprint scheint es zu klappen. [blueprint](https://github.com/danielperna84/custom_homematic/blob/devel/blueprints/automation/homematicip_local_reactivate_device_full.yaml)

**Comment by kwithus** [comment]
Ich habe die Erfahrung gemacht, dass es unterschiedliche Szenarien zu geben scheint. Manchmal (meist nach HA Neustart) erscheint ein HM-device nur in HA als nicht verfügbar, dann hilft das blueprint eigentlich immer. 
manchmal wird ein HM-device aber auch in Homematic als nicht erreichbar angezeigt, ist dann aber in dem Moment wieder verfügbar, wo ich es in der HM WebUI einmal betätige. Dabei reicht es zum Beispiel einen Schalter, der aktuell ausgeschaltet ist, in der WebUI nochmal auszuschalten. Das scheint n der TaAt ein Kommunikationsproblem innerhalb von Homematic zu sein. Da hilft das blueprint dann auch nicht, sondern nur wie geschildert das Betätigen des Geräts.
Da das Gerät dann in HA nicht verfügbar ist, lässt es sich von dort wohl kaum lösen.

**Reply by SukramJ** [reply]
Ein **einzelnes** Gerät ist in HA nur `unavailable` wenn die CCU ein `UNREACH` Event sendet, d.h die CCU informiert über ein Problem mit einem Gerät. 
Mit dem `homematicip_local.force_device_availability` oder den Blueprints betreiben wir eigentlich nur "Augenwischerei", den das Problem besteht zwischen der CCU und dem phyischem Gerät.

**Reply by emufan** [reply]
Diese Augenwischerei funktioniert aber perfekt. 👍 

> manchmal wird ein HM-device aber auch in Homematic als nicht erreichbar angezeigt, ist dann aber in dem Moment wieder verfügbar, wo ich es in der HM WebUI einmal betätige. Dabei reicht es zum Beispiel einen Schalter, der aktuell ausgeschaltet ist, in der WebUI nochmal auszuschalten. Das scheint n der TaAt ein Kommunikationsproblem innerhalb von Homematic zu sein. Da hilft das blueprint dann auch nicht, sondern nur wie geschildert das Betätigen des Geräts.

Doch. Zumindest bei mir. Das war ja die Ausgangslage. Ich habe diese Fälle und musste dann in HA warten, bis wieder erreichbar und aktiv in HA. In WebUI und am Hardwaretaster konnte ich aber drücken, die Verbindung wurde dann wieder hergestellt und das Gerät wurde bedient. Und eben das klappt (bei mir) nun auch über diese Automatisierungen. Ich nutze die hier (nicht als Blueprint, ist aber ja wurscht) für Devicetype BROLL. Und kann auch bei eigentlicher/angezeigter Unerreichbarkeit immer direkt wieder erfolgreich bedienen, als wäre ich am Hardwaretaster des BROLLs.

**Reply by kwithus** [reply]
Ich vermute, dass die Fehlermeldung in Homematic durch Unterschiedliche Dinge ausgelöst wird und es daher manchmal klappt und manchmal nicht. Jedenfalls scheint es prinzipiell ein HM Problem zu sein, das vermutlich etwas mit Duty cycle und anderen Funkproblemen zu tun hat.

**Reply by emufan** [reply]
Klar. Kann sicher unterschiedliche Gründe geben. Glaube aber eher, dass es rein fehlerhaftes Statusmanagement in der CCU ist, fehlende Rückmeldeverarbeitung oder was auch immer. Wollte nur auf das 

> Da das Gerät dann in HA nicht verfügbar ist, lässt es sich von dort wohl kaum lösen.

eingehen. Weil dies tatsächlich der Grund dieser Erweiterung/Verbesserung/Augenwischerei ist. Damit klappt's zumindest hier immer, obwohl HA und WebUI eigentlich meinen, dass das Ding gar nicht da ist.
